src/subdomains/generic/user/models/auth/dto/auth-credentials.dto.ts

```diff
@@ -14,4 +14,10 @@ export class AuthCredentialsDto {
   @IsString()
   @Matches(GetConfig().signatureFormat)
   signature: string;
+
+  @ApiProperty()
+  @IsOptional()
+  @IsString()
+  @Matches(GetConfig().keyFormat)
+  key?: string;
```

Comment:
Theoretically, we could use conditional validation:
```ts
  @ApiPropertyOptional()
  @IsRequired()
  @IsString()
  @Matches(GetConfig().keyFormat)
  @ValidateIf((dto: AuthCredentialsDto) => /stake[a-z0-9]{54}/.test(dto.address))
  key?: string;
```

Reply:
Or make address methods (`isCardanoAddress`) on CryptoService `public static` and use them here.

Reply:
`@IsRequired()` can't be found